src/lib/support/jsontlv/TlvToJson.cpp

```diff
@@ -0,0 +1,371 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "lib/support/CHIPMemString.h"
+#include "lib/support/ScopedBuffer.h"
+#include <json/json.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/Base64.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/jsontlv/ElementTypes.h>
+#include <lib/support/jsontlv/TlvToJson.h>
+
+namespace chip {
+
+namespace {
+
+/*
+ * Encapsulates the different types of keys permissible.
+ *
+ * Root Key = Key with a name of 'value'. This is the top-most key in a given JSON object generated from TLV.
+ * Struct Field = Key containing the 32-bit field ID of an item in a struct.
```

Comment:
But struct field ids are 8-bit....

Reply:
32-bit tag number is possible for Common Profile Tag, which is supported by this implementation.

Reply:
Ah, ok.  Data model payloads don't do that, but fine.